dart/optimizer/GradientDescentSolver.cpp

```diff
@@ -0,0 +1,482 @@
+{
+  mGradientP.mMaxAttempts = _maxAttempts;
+}
+
+//==============================================================================
+size_t GradientDescentSolver::getMaxAttempts() const
+{
+  return mGradientP.mMaxAttempts;
+}
+
+//==============================================================================
+void GradientDescentSolver::setPerturbationStep(size_t _step)
+{
+  mGradientP.mPerturbationStep = _step;
+}
+
+//==============================================================================
+size_t GradientDescentSolver::getPerturbationStep() const
+{
+  return mGradientP.mPerturbationStep;
+}
+
+//==============================================================================
+void GradientDescentSolver::setMaxPerturbationFactor(double _factor)
+{
+  mGradientP.mMaxPerturbationFactor = _factor;
+}
+
+//==============================================================================
+double GradientDescentSolver::getMaxPerturbationFactor() const
+{
+  return mGradientP.mMaxPerturbationFactor;
+}
+
+//==============================================================================
+void GradientDescentSolver::setDefaultConstraintWeight(double _newDefault)
+{
+  mGradientP.mDefaultConstraintWeight = _newDefault;
+}
+
+//==============================================================================
+double GradientDescentSolver::getDefaultConstraintWeight() const
+{
+  return mGradientP.mDefaultConstraintWeight;
+}
+
+//==============================================================================
+Eigen::VectorXd& GradientDescentSolver::getEqConstraintWeights()
+{
+  return mGradientP.mEqConstraintWeights;
+}
+
+//==============================================================================
+const Eigen::VectorXd& GradientDescentSolver::getEqConstraintWeights() const
+{
+  return mGradientP.mEqConstraintWeights;
+}
+
+//==============================================================================
+Eigen::VectorXd& GradientDescentSolver::getIneqConstraintWeights()
+{
+  return mGradientP.mIneqConstraintWeights;
+}
+
+//==============================================================================
+const Eigen::VectorXd& GradientDescentSolver::getIneqConstraintWeights() const
+{
+  return mGradientP.mIneqConstraintWeights;
+}
+
+//==============================================================================
+void GradientDescentSolver::randomizeConfiguration(Eigen::VectorXd& _x)
+{
+  if(nullptr == mProperties.mProblem)
+    return;
+
+  if(_x.size() < static_cast<int>(mProperties.mProblem->getDimension()))
+    _x = Eigen::VectorXd::Zero(mProperties.mProblem->getDimension());
+
+  for(int i=0; i<_x.size(); ++i)
+  {
+    double lower = mProperties.mProblem->getLowerBounds()[i];
+    double upper = mProperties.mProblem->getUpperBounds()[i];
+    double step = upper - lower;
+    if(step > mGradientP.mMaxRandomizationStep)
+    {
+      step = 2*mGradientP.mMaxRandomizationStep;
+      lower = _x[i] - step/2.0;
```

Comment:
Yep, I was going to say that they would almost certainly compile to identical machine code, so we don't need to worry about it.